tests/test_attributes.py

```diff
@@ -0,0 +1,96 @@
+import httpx
+import pytest
+from fastapi import status
+
+from app.api import utility as util
+
+
+def test_partially_failed_terms_fetching_handled_gracefully(
+    test_app, monkeypatch
+):
+    """
+    Test that when getting term instances for an attribute (/attribute/{data_element_URI}) fails for some nodes due to an error,
+    the overall API get request still succeeds, and the response includes a list of the encountered errors along with the successfully fetched terms.
+    """
+    monkeypatch.setattr(
+        util,
+        "FEDERATION_NODES",
+        {
+            "https://firstpublicnode.org/": "First Public Node",
+            "https://secondpublicnode.org/": "Second Public Node",
+        },
+    )
+
+    mocked_node_attribute_response = {
+        "nb:Assessment": [
+            {
+                "TermURL": "cogatlas:trm_56a9137d9dce1",
+                "Label": "behavioral approach/inhibition systems",
+            },
+            {
+                "TermURL": "cogatlas:trm_55a6a8e81b7f4",
+                "Label": "Barratt Impulsiveness Scale",
+            },
+        ]
+    }
+
+    async def mock_httpx_get(self, **kwargs):
+        if (
+            kwargs["url"]
+            == "https://firstpublicnode.org/attributes/nb:Assessment"
+        ):
+            return httpx.Response(
+                status_code=200,
+                json=mocked_node_attribute_response,
+            )
+        return httpx.Response(
+            status_code=500, json={}, text="Some internal server error"
+        )
```

Comment:
:cherries: functionally the same, but I find it easier to read if the `if` statement just covers the fail / odd-one-out, and then the else path is the default / valid response